lib/js/src/manager/SdlManager.js

```diff
@@ -314,23 +315,29 @@ class SdlManager extends _SdlManagerBase {
     start () {
         if (this._lifecycleManager === null) {
             if (this.getAppConfig() !== null) {
-                if (this.getAppConfig().getTransportConfig() === null) {
-                    throw new Error('AppConfig must have a transport config defined');
+                this._lifecycleConfig = this.getAppConfig().getLifecycleConfig();
+
+                if (this._lifecycleConfig === null) {
+                    throw new Error('AppConfig must have a LifecycleConfig defined');
+                }
+
+                if (this._lifecycleConfig.getTransportConfig() === null) {
+                    throw new Error('LifecycleConfig must have a transport config defined');
                 }
 
-                if (this.getAppConfig().getAppName() === null) {
-                    throw new Error('AppConfig must have an app name defined');
+                if (this._lifecycleConfig.getAppName() === null) {
+                    throw new Error('LifecycleConfig must have an app name defined');
                 }
 
-                if (this.getAppConfig().getAppId() === null) {
-                    throw new Error('AppConfig must have an app ID defined');
+                if (this._lifecycleConfig.getAppId() === null) {
+                    throw new Error('LifecycleConfig must have an app ID defined');
                 }
 
                 if (this.getManagerListener() === null) {
                     throw new Error('An SdlManagerListener must be defined');
                 }
 
-                this._lifecycleManager = new _LifecycleManager(this.getAppConfig(), this._lifecycleListener);
+                this._lifecycleManager = new _LifecycleManager(this._lifecycleConfig, this._lifecycleListener);
```

Comment:
The LifecycleManager is being given a lifecycleConfig now, but the LifecycleManager file still refers to it as an AppConfig. This could make things confusing for the developer.